include/emds/emds.h

```diff
@@ -18,6 +18,7 @@
 #include <sys/types.h>
 #include <zephyr/sys/util.h>
 #include <zephyr/sys/slist.h>
+#include "../../subsys/emds/emds_flash.h"
```

Comment:
This should not be included this way. Here is something wrong in the design when an external h file is dependent on some internal h file. We need to move around on the emds_entry_status if this should work. Suddenly the application will be also dependent on emds_flash.h and that should not be the case.

Reply:
Yes, I saw that this was a bit strange. It should be solvable but Im not quite sure how to do it.